arbor/util/partition.hpp

```diff
@@ -52,7 +52,7 @@ class partition_range: public range<partition_iterator<I>> {
     void validate() const {
         auto ok = is_valid();
         if (!ok) {
-            throw invalid_partition(ok.second());
+            throw invalid_partition(ok.error());
```

Comment:
I am always happy when I see ambiguous `first` and `second` get replaced with meaningful names.